src/core/layertree/qgslayertreemodellegendnode.h

```diff
@@ -277,6 +280,28 @@ class CORE_EXPORT QgsSymbolLegendNode : public QgsLayerTreeModelLegendNode
      */
     void setTextOnSymbolTextFormat( const QgsTextFormat &format ) { mTextOnSymbolTextFormat = format; }
 
+    /**
+     * Returns the legend label of the layer or symbol.
+     * \since QGIS 3.8
+     */
+    QString getCurrentLabel() const;
+
+    /**
+     * Create an expression context containing symbol elements
+     * \param context extra QgsExpressionContext to use for evaluation
+     * \since QGIS 3.8
+     */
+    QgsExpressionContext createExpressionContext( QgsExpressionContext context = QgsExpressionContext() ) const;
+
+    /**
+    * Evaluate the label with the proper context
+    * \param label text to evaluate
+    * \param context extra QgsExpressionContext to use for evaluation
+    * \since QGIS 3.8
+    */
+    QString evaluateLabelExpression( QString label, QgsExpressionContext context = QgsExpressionContext() ) const;
```

Comment:
Should this be private?

Reply:
I'm neutral about hiding this. If you want I can move it.

Reply:
I can set it to private if you feel it should be, I'm neutral about this change.

Reply:
Basically, make everything that's only needed private.

Reply:
Done!